File: Demo/Sources/ViewControllers/DataView/Simple/SimpleDataViewController.m
//
//  Globus
//

#import "SimpleDataViewController.h"
#import "SimpleDataViewCell.h"
#import "SimpleDataProvider.h"

@interface SimpleDataViewController ()
@end

static NSString* SimpleCellIdentifier = @"Simple";

@implementation SimpleDataViewController

#pragma mark - Init / Free

- (void)setup {
    [super setup];
    
    self.edgesForExtendedLayout = UIRectEdgeNone;
    self.title = @"SimpleDataView";
    
    self.provider = [SimpleDataProvider dataProvider];
    self.spinnerView = [GLBArcSpinnerView new];
    self.spinnerView.color = UIColor.blueColor;
}

#pragma mark - GLBViewController

- (NSArray< UIBarButtonItem* >*)prepareNavigationLeftBarButtons {
    return @[
        [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"MenuButton"] style:UIBarButtonItemStylePlain target:self action:@selector(pressedMenu:)]
    ];
}

#pragma mark - GLBSimpleDataViewController

- (GLBDataViewContainer< GLBSimpleDataViewControllerContentContainerProtocol >*)prepareContentContainer {
    return [SimpleDataViewControllerContentContainer container];
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

@end

@implementation SimpleDataViewControllerContentContainer

+ (NSDictionary< NSString*, Class >*)mapCells {
    return @{
        SimpleCellIdentifier: SimpleDataViewCell.class
    };
}

- (void)prepareWithModel:(id)model {
    [self appendIdentifier:SimpleCellIdentifier byData:model];
}

- (void)cleanupWithModel:(id)model {
    [self deleteAllItems];
}

@end

File: Demo/Sources/ViewControllers/DataView/Simple/SimpleDataViewController.swift
//
//  Globus
//

import Globus

class SimpleDataViewController: GLBSimpleDataViewController {
    
    // MARK: - Init / Free
    
    override func setup() {
        super.setup()
        
        self.edgesForExtendedLayout = UIRectEdge.init(rawValue: 0);
        self.title = "SimpleDataView"
        
        self.provider = SimpleDataProvider();
        self.spinnerView = GLBArcSpinnerView();
        self.spinnerView!.color = UIColor.blue;
    }
    
    // MARK: - GLBViewController
    
    override func prepareNavigationLeftBarButtons() -> [UIBarButtonItem] {
        return [
            UIBarButtonItem(image: UIImage(named: "MenuButton"), style: .plain, target: self, action: #selector(pressedMenu(_ :)))
        ]
    }
    
    // MARK: - GLBSimpleDataViewController
    
    override func prepareContentContainer() -> GLBDataViewContainer? {
        return SimpleDataViewControllerContentContainer()
    }
    
    // MARK: - Actions
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
}

class SimpleDataViewControllerContentContainer: GLBSimpleDataViewControllerContentContainer {
    
    // MARK: - GLBSimpleDataViewControllerContentContainer
    
    override class func mapCells() -> [String: AnyClass] {
        return [
            Const.SimpleCellIdentifier: SimpleDataViewCell.self
        ]
    }
    
    override func prepare(withModel model: Any) {
        self.appendIdentifier(Const.SimpleCellIdentifier, byData: model)
    }
    
    override func cleanup(withModel model: Any) {
        self.deleteAllItems()
    }
    
    // MARK: - Const
    
    struct Const {
        static let SimpleCellIdentifier = "Simple"
    }
    
}
